### PR TITLE
Feature login

### PR DESCRIPTION
**Pull request by DiegoCuevas**
***Implementation notes***

- I added a login function in rails and react.

### CODE REVIEW

**Comment by AngieCristina** [comment]
I recommend you to list all your implementation notes, to recognize yourr commit on time

File: Api/config/initializers/cors.rb [hunk 1]
```diff
@@ -1,16 +1,10 @@
-# Be sure to restart your server when you modify this file.
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
```
Comment (AngieCristina) [inline Api/config/initializers/cors.rb:1]: Could you tell me, what is Cors?

File: Api/db/migrate/20190710215625_create_users.rb [hunk 2]
```diff
@@ -3,11 +3,13 @@ def change
     create_table :users do |t|
       t.string :rol
       t.string :name
+      t.string :token
       t.string :lastName
       t.string :email
-      t.string :password
+      t.string :password_digest
```
Comment (AngieCristina) [inline Api/db/migrate/20190710215625_create_users.rb:9]: What is this `password_digest` means?

File: Api/spec/models/request_spec.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
-require 'rails_helper'
+# require 'rails_helper'
```
Comment (AngieCristina) [inline Api/spec/models/request_spec.rb]: Why do you comment all of those lines of code, Don't you think is much better to delete it?

File: Api/app/models/user.rb [hunk 4]
```diff
@@ -1,4 +1,16 @@
 class User < ApplicationRecord
   has_many :requester_request, class_name: 'Request', foreign_key: 'requester_id'
   has_many :requested_request, class_name: 'Request', foreign_key: 'requested_id'
+  has_secure_password
```
Comment (MariethPerez) [inline Api/app/models/user.rb:4]: Hi!!! Please Could you explain me how works 'has_secure_password'?

File: App/src/views/HomeView.js [hunk 5]
```diff
@@ -1,7 +1,15 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core";
 import React from "react";
+import { Redirect, navigate } from "@reach/router";
```
Comment (MariethPerez) [inline App/src/views/HomeView.js]: Why do you put 'navigate ' in this line? if you don't use them in this file.

File: App/src/views/HomeView.js [hunk 6]
```diff
@@ -1,7 +1,15 @@
...
+import { useUser, useUserUpdater } from "../contexts/user";
```
Comment (MariethPerez) [inline App/src/views/HomeView.js]: Why do you put 'useUserUpdater' in this line? if you don't use them in this file.

**Review by MariethPerez — DISMISSED** [review]

**Review by AngieCristina — APPROVED** [review]

**Review by MariethPerez — APPROVED** [review]

**Pull request merged**